### DISCUSSION TITLE
How to mass-edit custom fields?

### DISCUSSION THREAD

**Discussion by Trilliant-JP** [snipe/snipe-it#12345] [Q&A]
We have had to add two additional custom fields to a custom fieldset for laptops, and we need to populate all of them with the same thing in each field.

I added the 2 additional fields to the fieldset and then selected some of the endpoints that the fieldset applies to - but none of the custom fields show up in the edit screen.

Am I going to have to edit every asset individually to type in these values?  That's going to take days.  There HAS to be a way to mass-edit a custom field, right?

**Comment by corydlamb** [comment]
I think you should be able to do this through a CSV import, just grab all of the info on the devices you have and make sure you have the requisite fields in the CSV then add the custom fieldset to the CSV and mark them all with the same answer that you would like. There's some info on importing via CSV files [here](https://snipe-it.readme.io/docs/importing).

**Reply by Trilliant-JP** [reply]
Thanks @corydlamb I appreciate your answer.  I'm being told by my team that importing a CSV might create hundreds of duplicate assets if you're not careful.  So should we run a custom report with every field to CSV, mass add the 2 custom field values, and reimport?

**Reply by corydlamb** [reply]
> Thanks @corydlamb I appreciate your answer.  I'm being told by my team that importing a CSV might create hundreds of duplicate assets if you're not careful.  So should we run a custom report with every field to CSV, mass add the 2 custom field values, and reimport?

I believe that would work, if an asset is already in the database it shouldn't duplicate itself (I believe it goes by serial number so as long as that doesn't change I think you're in the clear.) Snipe actually has a very good backup feature that you can use before doing these massive CSV imports as well!
